src/app/components/shell/router.js

```diff
@@ -31,41 +32,51 @@ const Error404 = loadable({
     loader: () => import('~/pages/404/404'),
     loading: () => <h1>404</h1>
 });
-const DefaultLayout = loadable({
-    loader: () => import('~/layouts/default/default'),
-    loading: LoadingPlaceholder
-});
 
-function ImportedPage({name}) {
+function useLoading(name) {
     const {pathname} = useLocation();
-    const Page = React.useMemo(
-        () => {
-            function Loading({error, pastDelay, retry}) {
-                if (error) {
-                    if (error.code === 'MODULE_NOT_FOUND') {
-                        return pathname.endsWith('/') ? <Fallback name={name} />
-                        : <Navigate to={`${pathname}/`} replace />;
-                    }
-                    return <div>Error! <button onClick={ retry }>Retry</button></div>;
-                }
-                if (pastDelay) {
-                    return <LoadingPlaceholder />;
-                }
-                return null;
-            }
-
-            return loadable({
-                loader: () => import(`~/pages/${name}/${name}`),
-                loading: Loading,
-                render(loaded, props) {
-                    const Component = loaded.default;
 
-                    return <DefaultLayout><Component {...props} /></DefaultLayout>;
+    return React.useCallback(
+        ({error, pastDelay, retry}) => {
+            if (error) {
+                if (error.code === 'MODULE_NOT_FOUND') {
+                    return pathname.endsWith('/')
+                    ? <Fallback name={name} />
+                    : <Navigate to={`${pathname}/`} replace />;
                 }
-            });
+                return <div>Error! <button onClick={ retry }>Retry</button></div>;
+            }
+            if (pastDelay) {
+                return <LoadingPlaceholder />;
+            }
+            return null;
         },
         [name, pathname]
     );
+}
+
+function usePage(name) {
+    const loading = useLoading(name);
+    const {setLayoutParameters} = useLayoutContext();
+
+    return React.useMemo(
+        () => loadable({
+            loader: () => import(`~/pages/${name}/${name}`),
+            loading,
+            render(loaded, props) {
+                const Component = loaded.default;
+
+                setLayoutParameters({name: 'default'});
```

Comment:
this may cause flicker when navigating between two non-default layout pages, but to be fair i don't think we have that situation in reality right now

Reply:
I think if it gets to the `render` there, it's a default layout page, right?